x-pack/plugins/security_solution/server/lib/detection_engine/routes/index/signals_mapping.json

```diff
@@ -334,8 +334,15 @@
           "status": {
             "type": "keyword"
           },
-          "threshold_count": {
-            "type": "float"
+          "threshold_result": {
```

Comment:
For Indicator Match rules, I'm going to be adding similar fields to reflect the circumstances that generated the signal. I was thinking that the field would contain an array of objects, where each object represents a document in the threat index.

Do you think it makes sense to try and normalize these two use cases? I was thinking that beginning with

```json
  "matches": {
    "type": "object",
    "dynamic": "false"
  },
```
would give rules a place to at least _store_ this info, independent of it being queryable. It could potentially include `count` and `value` fields as you propose here, though, although I'd be hesitant to add mappings until we have a need for them (we don't currently use `threshold_count`, right?). 

For indicator matches I'm imagining a match looking something like

```js
{ 
  value: "my_value",
  field: "my_field",
  doc: "json string of doc in threat index"
}
```

For your use case I think we could keep e.g. `threshold_value` as a mapped field to be used in this dedup logic. Thoughts?

Reply:
@rylnd Couple thoughts...

1. I don't think this totally addresses the threshold rule case, but your thinking is similar to mine in a couple of ways... threshold rules have both a) a query, which would result in N matches like you have above (although I'm a bit unclear about the `doc` field...), and b) an optional bucket from the terms aggregation... there can also be additional filters on top of this. I think your solution would address `a`, but not `b`, as I need to be able to aggregate on the bucket values.

(more to come).

Reply:
We discussed this offline, and agreed that the two use cases should remain separate.